### ISSUE TITLE
Uncaught TypeError: Path must be a string. Received undefined

### ISSUE THREAD

**Issue by DeanmvSG** [file-icons/atom#512]
1. Open a project 
2. Use `cmd+p` to bring up fuzzy finder
3. Start typing letters (for me it triggers with `icon`, `controllers`, `appc` and others
4. Error is thrown

**Atom Version**: 1.13.0
**Electron Version**: 1.3.13
**System**: Mac OS X 10.12.2
**Thrown From**: [file-icons](https://github.com/file-icons/atom) package, v2.0.8

### Stack Trace

Uncaught TypeError: Path must be a string. Received undefined

```
At path.js:7

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at resolve (path.js:1148:7)
    at File.Resource (/Users/deanvizer/.atom/packages/file-icons/lib/filesystem/resource.js:29:10)
    at File (/Users/deanvizer/.atom/packages/file-icons/lib/filesystem/file.js:16:3)
    at FileSystem.get (/Users/deanvizer/.atom/packages/file-icons/lib/filesystem/filesystem.js:97:7)
    at FuzzyFinder.trackEntry (/Users/deanvizer/.atom/packages/file-icons/lib/consumers/consumer.js:177:27)
    at timeouts.set._ (/Users/deanvizer/.atom/packages/file-icons/lib/consumers/fuzzy-finder.js:155:10)
```

### Commands

```
     -0:05.6.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-ariake-dark-syntax.theme-nord-atom-ui)
     -0:00.8.0 fuzzy-finder:toggle-file-finder (atom-pane.pane.active)
```

### Config

```json
{
  "core": {
    "allowPendingPaneItems": false,
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.json": [
        ".arcconfig"
      ],
      "source.python": [
        "BUCK"
      ],
      "text.html.liquid": [
        "inc",
        "tpl",
        "htm",
        "html"
      ]
    },
    "disabledPackages": [
      "wrap-guide",
      "htmlhint",
      "Remote-FTP"
    ],
    "excludeVcsIgnoredPaths": false,
    "followSymlinks": false,
    "openEmptyEditorOnStart": false,
    "packagesWithKeymapsDisabled": [
      "sass-autocompile",
      "highlight-selected",
      "highlight-line"
    ],
    "telemetryConsent": "limited",
    "themes": [
      "nord-atom-ui",
      "ariake-dark-syntax"
    ],
    "useCustomTitleBar": true
  },
  "file-icons": {
    "forceShow": true
  }
}
```

### Installed Packages

```coffee
# User
ariake-dark-syntax, v0.0.6 (active)
atom-beautify, v0.29.16 (inactive)
atom-compile-less, v0.12.0 (inactive)
aurora-theme, v1.2.2 (inactive)
base64, v0.3.0 (inactive)
file-icons, v2.0.8 (active)
genesis-syntax, v1.0.7 (inactive)
genesis-ui, v0.5.0 (inactive)
github-atom-dark-syntax, v0.3.2 (inactive)
github-atom-light-syntax, v0.3.2 (inactive)
grunt-runner, v0.14.0 (active)
hide-tree, v0.0.20 (active)
highlight-line, v0.12.0 (active)
highlight-selected, v0.11.2 (active)
jshint, v1.8.6 (active)
language-liquid, v0.5.1 (active)
less-than-slash, v0.16.0 (active)
merge-conflicts, v1.4.4 (active)
nord-atom-syntax, v0.4.0 (inactive)
nord-atom-ui, v0.9.0 (active)
one-dark-vivid-syntax, v1.7.0 (inactive)
pigments, v0.38.0 (active)
project-manager, v3.3.3 (active)
react, v0.16.2 (active)
Remote-FTP, v0.9.4 (inactive)
sass-autocompile, v0.13.2 (active)
Siteglide-FTP, v1.1.3 (active)
sort-lines, v0.14.0 (inactive)
spacegray-atom-dark-syntax, v1.2.0 (inactive)
spacegray-dark-ui, v0.12.0 (inactive)
tab-title, v0.3.3 (active)
atom-dark-syntax, v0.28.0 (inactive)
atom-dark-ui, v0.53.0 (inactive)
atom-light-syntax, v0.29.0 (inactive)
atom-light-ui, v0.46.0 (inactive)
base16-tomorrow-dark-theme, v1.4.0 (inactive)
base16-tomorrow-light-theme, v1.4.0 (inactive)
one-dark-ui, v1.8.2 (inactive)
one-light-ui, v1.8.2 (inactive)
one-dark-syntax, v1.6.0 (inactive)
one-light-syntax, v1.6.0 (inactive)
solarized-dark-syntax, v1.1.1 (inactive)
solarized-light-syntax, v1.1.1 (inactive)
about, v1.7.2 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.14.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.33.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.29.0 (inactive)
autosave, v0.23.2 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.43.2 (active)
bracket-matcher, v0.85.1 (active)
command-palette, v0.39.1 (inactive)
deprecation-cop, v0.55.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.204.5 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.2.0 (active)
go-to-line, v0.31.2 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.1 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.159.1 (active)
metrics, v1.1.2 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.2 (inactive)
settings-view, v0.244.0 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.5 (active)
status-bar, v1.6.0 (active)
styleguide, v0.48.0 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.1 (active)
timecop, v0.33.2 (active)
tree-view, v0.211.1 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.39.0 (inactive)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.1 (active)
language-csharp, v0.13.0 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.47.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by Alhadis** [comment]
Run this in your console, please:

~~~js
var flog = require(atom.packages.activePackages["file-icons"].path + "/lib/log.js");
dir(flog.errors);
dir(flog.events);
~~~

Does either array contain anything insightful? Anything at all?

I'm not sure exactly what's going on, but it sounds ike there might be a broken symbolic link that's being followed where it shouldn't...

**Comment by DeanmvSG** [comment]
@Alhadis The second array is empty but the first one contains lots of references to `TREE ENTRY: Oddball - "this.resource.isSubmodule" expected to be true`

Screenshot of them below (some file paths omitted!) 
<img width="1792" alt="screen shot 2017-01-11 at 11 06 13" src="https://cloud.githubusercontent.com/assets/15265711/21846240/2978cdc8-d7ee-11e6-9616-53a4940861a0.png">

**Comment by DZPM** [comment]
Same bug here. Atom 1.12.9 in Ubuntu 16.04.

Similar output in the console. My `flog.errors` array has 26 objects, and the error message is the same: `"TREE ENTRY: Oddball - `this.resource.isSubmodule` expected to be true"`.

The error started after updating several atom packages.

**Comment by Alhadis** [comment]
Connected to #510; fixed by 09327d0.
